### PR TITLE
Override CloseFragmentDialog to check for nullity to avoid crashes

### LINKED ISSUE #2686: App crashes on restore from Edit TE screen when the duration popup is open
Steps to reproduce:

1 - Tap any existing TE from the main log
2 - Tap the Start/Stop areas to open the Edit Start/Stop/Duration popup.
3 - Press Home
4 - 💥

### PR DESCRIPTION
**Pull request by daividssilverio**
Closes #2686 

The source of the problem is most likely caused by a bug on the Mvx implementation used to close dialog fragments when sending the current activity to background.

What is happening is that by the time the presenter tries to close the DialogFragment used for the `SelectTimeFragment` (edit time popup), the references for the Activity and the FragmentManager are lost or set to null, causing the NullPointerException at Line 439 on the superclass `MvxAppCompatViewPresenter`.

```
var toClose = CurrentFragmentManager.FindFragmentByTag(tag);
```

Functional testing an actual devices is a must. Already tested on Nexus 5 (Android 23).

🦑 whenever

### CODE REVIEW

File: Toggl.Giskard/Presenters/TogglPresenter.cs [hunk 1]
```diff
@@ -50,5 +52,17 @@ public override void ChangePresentation(MvxPresentationHint hint)
 
             base.ChangePresentation(hint);
         }
+
+        protected override bool CloseFragmentDialog(IMvxViewModel viewModel, MvxDialogFragmentPresentationAttribute attribute)
+        {
+            string tag = FragmentJavaName(attribute.ViewType);
```
Comment (willsb) [inline Toggl.Giskard/Presenters/TogglPresenter.cs]: Why not `var`?

File: Toggl.Giskard/Presenters/TogglPresenter.cs [hunk 2]
```diff
@@ -50,5 +52,17 @@ public override void ChangePresentation(MvxPresentationHint hint)
...
+            var toClose = CurrentFragmentManager?.FindFragmentByTag(tag);
+            if (toClose != null && toClose is DialogFragment dialog)
```
Comment (willsb) [inline Toggl.Giskard/Presenters/TogglPresenter.cs]: The first null check is not needed

File: Toggl.Giskard/Presenters/TogglPresenter.cs [hunk 3]
```diff
@@ -50,5 +52,17 @@ public override void ChangePresentation(MvxPresentationHint hint)
...
+            {
+                dialog.DismissAllowingStateLoss();
+                return true;
+            }
+            return false;
```
Comment (willsb) [inline Toggl.Giskard/Presenters/TogglPresenter.cs]: One whitespace above this line, please

File: Toggl.Giskard/Presenters/TogglPresenter.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Giskard/Presenters/TogglPresenter.cs]: This was copy/paste from the Mvx source code. I'm fixing the style now.

File: Toggl.Giskard/Presenters/TogglPresenter.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Giskard/Presenters/TogglPresenter.cs]: Indeed.

**Pull request merged**